Fix BSR->Dense Batched Bug

amjames opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82121
* #82122
* #80781
* __->__ #82120
* #82119

A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general.

amjames committed:
Fix BSR->Dense Batched Bug

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"




[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"




[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames commented:
@bhosmer @nikitaved @pearu This is ready for review

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

nikitaved commented on aten/src/ATen/native/TensorConversions.cpp:
```diff
@@ -400,10 +400,10 @@ Tensor sparse_compressed_to_dense(
-      col_indices = col_indices.flatten(0, n_batch_dim);
-      values = values.flatten(0, n_batch_dim);
-      dense = dense.flatten(0, n_batch_dim);
+      crow_indices = crow_indices.flatten(0, n_batch_dim - 1);
```
alternatively, we can do `flatten(0, -2)` to avoid any explicit batch dims :)

nikitaved reviewed (approved):
Looks good!

amjames commented:
@pytorchbot merge

amjames commented:
@pytorchbot merge

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames committed:
Update on "Fix BSR->Dense Batched Bug"


A todo in the tests which should have been removed and addressed before the initial PR landed was left, and so left holes in testing BSR-> Dense. This addresses the underlying issue and removes the hole in test coverage. #8071 Introduces more comprehensive test coverage for sparse compressed <-> Dense conversion in general. 

[ghstack-poisoned]

amjames commented:
@pytorchbot merge
